src/mod_muc_api.erl

```diff
@@ -0,0 +1,375 @@
+
+-spec to_user_map(mod_muc_room:user(), boolean()) -> user_map().
+to_user_map(#user{role = Role, nick = Nick}, false = _WithJID) ->
+    #{jid => null, role => Role, nick => Nick};
+to_user_map(#user{jid = JID, role = Role, nick = Nick}, true) ->
+    #{jid => JID, role => Role, nick => Nick}.
+
+-spec room_to_item(tuple(), jid:lserver(), jid:jid()) -> {true, short_room_desc()} | false.
+room_to_item({{Name, _}, Pid}, MUCServer, From) ->
+     case catch gen_fsm_compat:sync_send_all_state_event(
+                  Pid, {get_disco_item, From, <<"en">>}, 100) of
+         {item, Desc} ->
+             Map = room_desc_to_map(Desc),
+             {true, Map#{jid => jid:to_binary({Name, MUCServer, <<>>})}};
+         _ ->
+             false
+     end.
+
+-spec room_desc_to_map(binary()) -> #{title := binary(), private => boolean(),
+                                      users_number => non_neg_integer()}.
+room_desc_to_map(Desc) ->
+    MP = "(\\S+)( \\((private, )?(\\d+)\\))?",
+    {match, [TitlePos, _, PrivatePos, NumberPos]} = re:run(Desc, MP, [{capture, [1, 2, 3, 4]}]),
+    Title = binary:part(Desc, TitlePos),
+    case NumberPos of
+        {-1, 0} ->
+            #{title => Title};
+        _ ->
+            Private = {-1, 0} =/= PrivatePos,
+            Number = binary_to_integer(binary:part(Desc, NumberPos)),
+            #{title => Title, private => Private, users_number => Number}
+    end.
+
+-spec verify_room(jid:jid(), jid:jid()) ->
+    ok | {internal | not_found, term()}.
+verify_room(BareRoomJID, OwnerJID) ->
+    case mod_muc_room:can_access_room(BareRoomJID, OwnerJID) of
+        {ok, true} ->
+            ok;
+        {ok, false} ->
+            {internal, "Room is locked"};
+        {error, not_found} ->
+            ?ROOM_NOT_FOUND_RESULT
+    end.
+
+iq(Type, Sender, Recipient, Children)
+  when is_binary(Type), is_list(Children) ->
+    Addresses = address_attributes(Sender, Recipient),
+    #xmlel{name = <<"iq">>,
+           attrs = Addresses ++ [{<<"type">>, Type}],
+           children = Children
+          }.
+
+message(Sender, Recipient, Type, Contents)
+  when is_binary(Type), is_list(Contents) ->
+    Addresses = address_attributes(Sender, Recipient),
+    Attributes = case Type of
+                     <<>> -> Addresses;
+                     _ -> [{<<"type">>, Type} | Addresses]
+                 end,
+    #xmlel{name = <<"message">>,
+           attrs = Attributes,
+           children = Contents
+          }.
+
+query(XMLNameSpace, Children)
+  when is_binary(XMLNameSpace), is_list(Children) ->
+    #xmlel{name = <<"query">>,
+           attrs = [{<<"xmlns">>, XMLNameSpace}],
+           children = Children
+          }.
+
+presence(Sender, Recipient) ->
+    #xmlel{name = <<"presence">>,
+           attrs = address_attributes(Sender, Recipient),
+           children = [#xmlel{name = <<"x">>,
+                              attrs = [{<<"xmlns">>, ?NS_MUC}]}]
+          }.
+
+declination(Sender, Recipient) ->
```

Comment:
Why is this called `declination`?

Reply:
This code was taken from the MUC Rest API. As `XEP-0045` says:
> If the initial room owner wants to accept the default room configuration (i.e., create an "instant room"), the room owner MUST decline an initial configuration form by sending an IQ [...]
So I think it's taken from "decline configuration".